src/converse-core.js

```diff
@@ -281,7 +281,7 @@
             }
         }
 
-        if(this.time_format !== 'h' && this.time_format !== 'H') {
+        if(this.time_format !== 'hh:mm' && this.time_format !== 'HH:MM' && this.time_format !== 'hh:mm a') {
```

Comment:
Remove this whole `if` statement. We don't want to restrict what format the user can provide.